src/components/grid/grid-item/grid-item.style.js

```diff
@@ -1,26 +1,33 @@
 import styled, { css } from 'styled-components';
+import { space, grid, flexbox } from 'styled-system';
 import PropTypes from 'prop-types';
 
 function responsiveGridItem(responsiveSettings) {
   return responsiveSettings.map((setting) => {
     const {
-      colStart,
-      colEnd,
-      maxWidth,
-      rowStart,
-      rowEnd,
       alignSelf,
-      justifySelf
+      gridColumn,
+      gridRow,
+      maxWidth,
+      justifySelf,
+      p,
+      pl,
+      pr,
+      pt,
+      pb
     } = setting;
 
     return css`
       @media screen and (max-width: ${maxWidth}) {
         align-self: ${alignSelf || 'stretch'};
         justify-self: ${justifySelf || 'stretch'};
-        grid-column-start: ${colStart};
-        grid-column-end: ${colEnd};
-        grid-row-start: ${rowStart};
-        grid-row-end: ${rowEnd};
+        grid-column: ${gridColumn};
```

Comment:
How come you're not just adding `${space}` and `${grid}` here?

Reply:
if I pass that, then it would place original values instead of these from the responsiveSetting object

Reply:
Ah ok I see! In that case I think these should support being able to be a number as well so e.g. 
```js
padding-left: ${typeof pl === 'number' ? theme.space[pl] : pl]};
```
Otherwise it won't be consistent with all the other uses of these spacing props which would be confusing I think